runtime/service/src/test/java/org/apache/polaris/service/events/jsonEventListener/aws/cloudwatch/AwsCloudWatchEventListenerTest.java

```diff
@@ -0,0 +1,297 @@
+  private static final String REALM = "test-realm";
+  private static final String TEST_USER = "test-user";
+  private static final Clock clock = Clock.systemUTC();
+
+  @Mock private AwsCloudWatchConfiguration config;
+
+  private ExecutorService executorService;
+  private AutoCloseable mockitoContext;
+
+  @BeforeEach
+  void setUp() {
+    mockitoContext = MockitoAnnotations.openMocks(this);
+    executorService = Executors.newSingleThreadExecutor();
+
+    // Configure the mocks
+    when(config.awsCloudwatchlogGroup()).thenReturn(LOG_GROUP);
+    when(config.awsCloudwatchlogStream()).thenReturn(LOG_STREAM);
+    when(config.awsCloudwatchRegion()).thenReturn("us-east-1");
+    when(config.synchronousMode()).thenReturn(false); // Default to async mode
+  }
+
+  @AfterEach
+  void tearDown() throws Exception {
+    if (mockitoContext != null) {
+      mockitoContext.close();
+    }
+    if (executorService != null) {
+      executorService.shutdownNow();
+      if (!executorService.awaitTermination(5, TimeUnit.SECONDS)) {
+        LOGGER.warn("ExecutorService did not terminate in time");
+      }
+    }
+    if (localStack.isRunning()) {
+      localStack.stop();
+    }
+  }
+
+  private CloudWatchLogsAsyncClient createCloudWatchAsyncClient() {
+    if (!localStack.isRunning()) {
+      localStack.start();
+    }
+    return CloudWatchLogsAsyncClient.builder()
+        .endpointOverride(localStack.getEndpoint())
+        .credentialsProvider(
+            StaticCredentialsProvider.create(
+                AwsBasicCredentials.create(localStack.getAccessKey(), localStack.getSecretKey())))
+        .region(Region.of(localStack.getRegion()))
+        .build();
+  }
+
+  private AwsCloudWatchEventListener createListener(CloudWatchLogsAsyncClient client) {
+    AwsCloudWatchEventListener listener =
+        new AwsCloudWatchEventListener(config, clock) {
+          @Override
+          protected CloudWatchLogsAsyncClient createCloudWatchAsyncClient() {
+            return client;
+          }
+        };
+
+    // Set up call context and security context
+    CallContext callContext = Mockito.mock(CallContext.class);
+    PolarisCallContext polarisCallContext = Mockito.mock(PolarisCallContext.class);
+    RealmContext realmContext = Mockito.mock(RealmContext.class);
+    SecurityContext securityContext = Mockito.mock(SecurityContext.class);
+    Principal principal = Mockito.mock(Principal.class);
+    when(callContext.getRealmContext()).thenReturn(realmContext);
+    when(callContext.getPolarisCallContext()).thenReturn(polarisCallContext);
+    when(realmContext.getRealmIdentifier()).thenReturn(REALM);
+    when(securityContext.getUserPrincipal()).thenReturn(principal);
+    when(principal.getName()).thenReturn(TEST_USER);
+    listener.callContext = callContext;
+    listener.securityContext = securityContext;
+
+    return listener;
+  }
+
+  @Test
+  void shouldCreateLogGroupAndStream() {
```

Comment:
We are missing the negative test here, we also ned to test that it doesn't fail if the log and group already exists. You could just call the create a second time